2016/proposals/singh-shubham.md

```diff
@@ -0,0 +1,210 @@
+This view will also list down the general guidelines for instructor training, informing what is required in each stage of the training.
+  * *ABOUT* , this view shows the information entered by the trainee in their initial application, they can also edit this information, and change passwords here.
+  * *NOTIFICATION*,  this view allows the trainee to view all the comments/feedback from the discussion leaders and the lesson maintainers.In the format *comment: by whom: timestamp*.
+
+
+## Technical Details
+The Software stack for this project can be divided into:
+
+ * Backend  : [Django](https://www.djangoproject.com/)
+ * Database : [SQLite](https://www.sqlite.org/)
+ * Frontend : [Bootstrap](http://getbootstrap.com/),[JQuery](https://jquery.com/) for AJAX requests
+
+### Features and Model Schemes
+
+Trainee would be the main Model, maintaining most of the Flags for Progress and other information:
+
+```python
+lass Trainee(AbstractBaseUser):
+    '''Model for a Trainee '''
+    # listing just essential fields, more fields can and should be added when implementing
+    COURSE_CHOICES = (
+        ('1', 'SWC'),
+        ('2', 'DC'),
+        )
+    personal    = models.CharField(max_length=STR_LONG)
+    middle      = models.CharField(max_length=STR_LONG, null=True, blank=True)
+    family      = models.CharField(max_length=STR_LONG)
+    email       = models.CharField(max_length=STR_LONG, unique=True, null=True, blank=True)
+    course      = models.CharField(max_length=STR_LONG, choices=COURSE_CHOICES, null=False, default='1')
+    group       = models.ForeignKey(Trainee_Group,default=None,null=True)
+    Attendance_Flag = models.BooleanField(default=False) # maintains attendance record
+    Discussion_Flag = models.BooleanField(default=False) # maintains PASS or FAIL
+    HomeWork_Flag = models.BooleanField(default=False) # maintains PASS or FAIL
+
+    username    = models.CharField(
+        max_length=STR_MED, unique=True,
+        validators=[RegexValidator(r'^[\w\-_]+$', flags=re.A)],
+    )
+ ```
+
+Maintaining Group information:
+
+```python
+class Trainee_Group(models.Model):
+
+    instructor_assigned=models.ForeignKey(Person)
+    primary_contact_personal    = models.CharField(max_length=STR_LONG)
+    primary_contact_middle      = models.CharField(max_length=STR_LONG, null=True, blank=True)
+    primary_contact_family      = models.CharField(max_length=STR_LONG)
+    primary_contact_email       = models.CharField(max_length=STR_LONG, unique=True, null=True, blank=True)
+    training_course             = models.CharField(max_length=STR_LONG)
+```
+Commenting System:
+Since comments/feedback are only meant to be unidirectional (i.e from the discussion_leaders, lesson_maintainers to the trainee) a simple Commenting Model would suffice and there is no need for third party apps.
```

Comment:
@gvwilson 

The Feedback/Comments are important information which can be archived, they can indicate why and why not a trainee was passed during any stage in training process.
I have added another View in Trainee Home, MESSAGES, which the trainees can use to converse with any of the instructors(these are private messages). We would be using [django-postman](http://django-postman.readthedocs.org/en/latest/quickstart.html) for this feature.